packages/@azure/servicebus/data-plane/test/sessionsTests.spec.ts

```diff
@@ -0,0 +1,494 @@
+  });
+
+  it("Unpartitioned Topics and Subscription with Sessions - Streaming Receiver: complete() removes message", async function(): Promise<
+    void
+  > {
+    await testComplete_streaming(unpartitionedTopicSessionClient, unpartitionedSubscriptionSessionClient);
+  });
+});
+
+describe("Accept a session by passing non-existing sessionId - Queue/Subscription has messages belonging to same sessionId", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await afterEachTest();
+  });
+
+  async function test_batching(
+    senderClient: QueueClient | TopicClient,
+    sessionClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+
+    await senderClient.send(testMessagesWithSessions[0]);
+
+    let receiverClient = await sessionClient.acceptSession({sessionId:"non"+testSessionId});
+    let msgs = await receiverClient.receiveBatch(1,10);
+    should.equal(msgs.length, 0);
+
+    receiverClient = await sessionClient.acceptSession({sessionId: testSessionId});
+    await testPeekMsgsLength(receiverClient, 1);
+    msgs = await receiverClient.receiveBatch(1);
+    should.equal(Array.isArray(msgs), true);
+    should.equal(msgs[0].body, testMessagesWithSessions[0].body);
+    should.equal(msgs[0].messageId, testMessagesWithSessions[0].messageId);
+    await msgs[0].complete();
+    await testPeekMsgsLength(receiverClient, 0);
+  }
+
+
+  it("Partitioned Queues with Sessions - Batch Receiver: no messages received", async function(): Promise<
+    void
+  > {
+    await test_batching(partitionedQueueSessionClient, partitionedQueueSessionClient);
+  });
+
+  it("Partitioned Topics and Subscription with Sessions - Batch Receiver: no messages received", async function(): Promise<
+    void
+  > {
+    await test_batching(partitionedTopicSessionClient, partitionedSubscriptionSessionClient);
+  });
+
+  it("Unpartitioned Queues with Sessions - Batch Receiver: no messages received", async function(): Promise<
+    void
+  > {
+    await test_batching(unpartitionedQueueSessionClient, unpartitionedQueueSessionClient);
+  });
+
+  it("Unpartitioned Topics and Subscription with Sessions - Batch Receiver: no messages received", async function(): Promise<
+    void
+  > {
+    await test_batching(unpartitionedTopicSessionClient, unpartitionedSubscriptionSessionClient);
+  });
+
+  async function test_streaming(
+    senderClient: QueueClient | TopicClient,
+    sessionClient: QueueClient | SubscriptionClient
+  ): Promise<void> {
+
+    await senderClient.send(testMessagesWithSessions[0]);
+
+    let receiverClient = await sessionClient.acceptSession({sessionId:"non"+testSessionId});
+    let receivedMsgs: ServiceBusMessage[] = [];
+    receiverClient.receive((messageSession: MessageSession, msg: ServiceBusMessage) => {
+      receivedMsgs.push(msg);
+      return Promise.resolve();
+    }, unExpectedErrorHandler);
+    should.equal(receivedMsgs.length, 0);
```

Comment:
We always wait a while after starting the streaming receiver, but we arent doing so here.
This check will always be true, because it will run before the callback is ever called.

Reply:
Yes, updated this.